### PR TITLE
Add import eks cluster test case

### PR DESCRIPTION
**Pull request by aaronRancher**
This PR tests importing an EKS cluster into Rancher.

### CODE REVIEW

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 1]
```diff
@@ -99,6 +101,43 @@ def test_create_hosted_eks_cluster_2():
...
+
+def create_resources_eks():
+    cluster_name = resource_prefix + "-ekscluster"
+    AmazonWebServices().create_eks_cluster(cluster_name)
+    AmazonWebServices().wait_for_eks_cluster_state(cluster_name, "ACTIVE")
+    return cluster_name
+
+
 @pytest.fixture(scope='module', autouse="True")
 def create_project_client(request):
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_hosted_eks_cluster.py:142]: Could we also have this cluster deleted ? From Rancher and from the EKS console after the test is done?

Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py:142]: The cluster already deleted from Rancher. I will look into how we can delete from EKS console.

Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py:143]: Addressed changes. The cluster will now be deleted from the EKS console.

File: tests/validation/lib/aws.py [hunk 2]
```diff
@@ -271,6 +271,19 @@ def delete_node(self, node, wait_for_deleted=False):
...
 
+    def delete_eks_cluster(self, cluster_name):
+        ng_names = self._eks_client.list_nodegroups(clusterName=cluster_name)
+        for node_group in ng_names['nodegroups']:
+            print("deleting node group: " + node_group)
+            delete_ng_response = self._eks_client.delete_nodegroup(
+                                                    clusterName=cluster_name,
+                                                    nodegroupName=node_group)
+        waiter_ng = self._eks_client.get_waiter('nodegroup_deleted')
+        for node_group in ng_names['nodegroups']:
```
Comment (izaac) [inline tests/validation/lib/aws.py:282]: Can we print a wait message here to help debugging if we reach a case where the waiter failed after attempts are exceeded? a print maybe ? or enclosing this in try-except of `WaiterError`.

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 3]
```diff
@@ -64,7 +66,7 @@ def test_create_hosted_eks_cluster_1():
 
 
 @ekscredential
-def test_create_hosted_eks_cluster_2():
+def test_eks_v2_create_hosted_cluster_2():
```
[the remark this answers is not in the document]
Comment (sowmyav27) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py:69]: This is related to the new EKS KE v2

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 4]
```diff
@@ -99,6 +101,44 @@ def test_create_hosted_eks_cluster_2():
...
+        "region": EKS_REGION,
+        "secretsEncryption": False,
+        "type": "eksclusterconfigspec"
+    }
+
+    cluster_config = {
+        "eksConfig": eks_config_temp,
+        "name": cluster_name,
+        "type": "cluster",
+        "dockerRootDir": "/var/lib/docker",
```
[the remark this answers is not in the document]
Comment (sowmyav27) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py:125]: This is what we see in the UI when we make a cluster create call.

File: tests/validation/lib/aws.py (hunk 2 above)
Comment (aaronRancher) [reply tests/validation/lib/aws.py:282]: Added a print statement. Boto waiter will throw an error if maximum tries are exceeded.

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 5]
```diff
@@ -99,6 +101,44 @@ def test_create_hosted_eks_cluster_2():
...
+        "enableNetworkPolicy": False,
+        "enableClusterAlerting": False,
+        "enableClusterMonitoring": False
+    }
+    create_and_validate_eks_cluster(cluster_config,
+                                    imported=True)
+    AmazonWebServices().delete_eks_cluster(cluster_name=display_name)
```
Comment (bmdepesa) [inline tests/validation/tests/v3_api/test_hosted_eks_cluster.py:132]: Delete the cluster in a finalizer so if `create_and_validate_eks_cluster` fails after the cluster is created, it still gets cleaned up.

File: tests/validation/lib/aws.py [hunk 6]
```diff
@@ -271,6 +271,19 @@ def delete_node(self, node, wait_for_deleted=False):
...
+            waiter_ng.wait(clusterName=cluster_name, nodegroupName=node_group)
+        print("deleting cluster: "+ cluster_name)
+        delete_response = self._eks_client.delete_cluster(name=cluster_name)
```
[the remark this answers is not in the document]
Comment (aaronRancher) [reply tests/validation/lib/aws.py:286]: Addressed in changes. `delete_response` is returned.

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py (hunk 5 above)
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py]: Addressed in changes. Cluster(s) are deleted in finalizer of `create_project_client`.

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 7]
```diff
@@ -99,27 +102,74 @@ def test_create_hosted_eks_cluster_2():
...
+    }
+    create_and_validate_eks_cluster(cluster_config,
+                                    imported=True)
+
+
+def create_resources_eks():
+    cluster_name = resource_prefix + "-ekscluster"
+    AmazonWebServices().create_eks_cluster(cluster_name)
+    AmazonWebServices().wait_for_eks_cluster_state(cluster_name, "ACTIVE")
+    IMPORTED_EKS_CLUSTERS.append(cluster_name)
```
Comment (bmdepesa) [inline tests/validation/tests/v3_api/test_hosted_eks_cluster.py:139]: Can we add the cluster name to the array before waiting for state? If the state doesn't get to the expected state before timeout, the cluster will not be deleted.

Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py]: Addressed in changes. Cluster name is added before waiting for state.

File: tests/validation/tests/v3_api/test_hosted_eks_cluster.py [hunk 8]
```diff
@@ -110,16 +150,23 @@ def fin():
...
     print("\nEKS Configuration: {}".format(cluster_config))
     cluster = client.create_cluster(cluster_config)
     print(cluster)
     cluster_details[cluster["name"]] = cluster
-    cluster = validate_cluster(client, cluster, check_intermediate_state=True,
-                               skipIngresscheck=True,
-                               timeout=DEFAULT_TIMEOUT_EKS)
+    if imported:
+        cluster = validate_cluster(client, cluster,
+                                   check_intermediate_state=False,
```
[the remark this answers is not in the document]
Comment (sowmyav27) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py:165]: Maybe we could have it like this
```
intermediate_state = False if imported else intermediate_state = True
cluster = validate_cluster(client, cluster,
                                   check_intermediate_state=intermediate_state,
                                   skipIngresscheck=True,
                                   timeout=DEFAULT_TIMEOUT_EKS)
```

[the remark this answers is not in the document]
Comment (izaac) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py:165]: I like that approach.

[the remark this answers is not in the document]
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py]: Addressed in changes.

[the remark this answers is not in the document]
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_hosted_eks_cluster.py]: Addressed in changed. intermediate_state is declared in 1 line.

**Pull request merged**